tensorflow_serving/model_servers/main.cc

```diff
@@ -128,15 +129,24 @@ tensorflow::Status LoadCustomModelConfig(
 }
 
 ModelServerConfig BuildSingleModelConfig(const string& model_name,
-                                         const string& model_base_path) {
+                                         const string& model_base_path,
+                                         const string& model_version_policy) {
   ModelServerConfig config;
   LOG(INFO) << "Building single TensorFlow model file config: "
             << " model_name: " << model_name
-            << " model_base_path: " << model_base_path;
+            << " model_base_path: " << model_base_path
+            << " model_version_policy: " << model_version_policy;
   tensorflow::serving::ModelConfig* single_model =
       config.mutable_model_config_list()->add_config();
   single_model->set_name(model_name);
   single_model->set_base_path(model_base_path);
+
+  FileSystemStoragePathSourceConfig_VersionPolicy parsed_policy;
+  bool valid_policy = FileSystemStoragePathSourceConfig_VersionPolicy_Parse(
+    model_version_policy, &parsed_policy);
+  if (valid_policy) {
```

Comment:
Please throw an error if it fails to parse. (We'll put the parsing code outside of this function.)